state/machine.go

```diff
@@ -2205,29 +2205,62 @@ func (m *Machine) RemoveUpgradeSeriesLock() error {
 
 // UpgradeSeriesStatus returns the status of a series upgrade.
 func (m *Machine) UpgradeSeriesStatus() (string, error) {
-	return "preparing", nil
+	coll, closer := m.st.db().GetCollection(machineUpgradeSeriesLocksC)
+	defer closer()
+
+	var lock upgradeSeriesLock
+	err := coll.FindId(m.Id()).One(&lock)
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+	s := fmt.Sprintf("%d", lock.PrepareUnits[0].Status)
```

Comment:
Suggest we rename to UpgradeSeriesPrepareStatus() and return the lock.PrepareStatus, and have a new Unit.UpgradeSeriesPrepareStatus() for the individual units?